src/db.ts

```diff
@@ -749,7 +749,6 @@ function validateDatabaseName(databaseName: string) {
   const invalidChars = [' ', '.', '$', '/', '\\'];
   for (let i = 0; i < invalidChars.length; i++) {
     if (databaseName.indexOf(invalidChars[i]) !== -1)
-      // TODO(NODE-3405): Change this out for a child of MongoParseError
       throw new MongoDriverError(
```

Comment:
why did we remove the todo without updating the error?

Reply:
We went through and removed the TODOs here since they have already been replaced in NODE-3405

Reply:
Please leave the TODO in this PR, the PR that addresses the comment should be the one to remove the line

Reply:
This TODO got moved up outside of the function and retagged with the right ticket. Refactoring `validateDatabaseName` falls under NODE-3484.

Reply:
The TODO here got updated to point to the correct ticket, [NODE-3484](https://jira.mongodb.org/browse/NODE-3484), and moved to the top of this function since the ticket calls for the refactoring of the entire thing.